### DISCUSSION TITLE
Resolving relations in graphql

### DISCUSSION THREAD

**Discussion by adamalfredsson** [prisma/prisma#11715] [Q&A] [answered]
I'm building my first application with Prisma and NestJS and I'm not confident in what strategy for resolving relations is appropriate. I'm following the [nest-js-prisma-starter](https://github.com/notiz-dev/nestjs-prisma-starter) as a guideline. In it, there is a `UserResolver` that has a field resolver for including posts ([code](https://github.com/notiz-dev/nestjs-prisma-starter/blob/main/src/resolvers/user/user.resolver.ts)):

```typescript
class UserResolver {
  constructor(
    private userService: UserService,
    private prisma: PrismaService
  ) {}
  
  // ...

  @ResolveField('posts')
  posts(@Parent() author: User) {
    return this.prisma.user.findUnique({ where: { id: author.id } }).posts();
  }
}
```

I have some questions regarding this:

1. Coming from TypeORM, I'm used to using a data-loader or something to batch all database queries made in field resolvers (for example if we're fetching many users with posts). Is this necessary with prisma or is it somehow handled internally?
2. Is it generally encouraged to use Prisma from a service rather than the resolver? Sometimes it feels like I'm duplicating the prisma interface in the service interface and an unnecessary abstraction.
3. Instead of joining relations on a field resolver level, is it an option to include the relation when doing the initial database query? When is it recommended to use which?

Thank you!

**Comment by Luke265** [comment, marked as answer]
You might want to take a look at [typegraphql-prisma](https://www.npmjs.com/package/typegraphql-prisma) or [prisma-nestjs-graphql](https://www.npmjs.com/package/prisma-nestjs-graphql).

In short, these packages will generate everything for you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
